### PR TITLE
Restructuring .NET samples repo

### PR DESCRIPTION
**Pull request by joaopgrassi**
This PR introduces changes in the structure of the .NET sample solution. In summary:

- Removed old console projects (`c#` and `vb`) that did the authorization_code flow.
- Introduced two new console projects for demonstrating the `client_credentials` flow (more suitable for console apps)
- Introduced an MVC app to demonstrate the `authorization_code` with PKCE
- Modified the readme.md's accordingly.

### CODE REVIEW

File: src/ConsoleClientCredentials.NETCore/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿# Tapkey API Console Sample with Client Credentials (C#, .NET Core 2.2) 
+
+## Client credentials flow
+This console application demonstrates how to consume the Tapkey API from an application using the client credentials flow. 
+
+The Client Credentials flow is tipically used when the client (the application) is acting on its own behalf. This means that there's no actual user/person interaction in the process. The client (application) is the owner of the resources.
```
Comment (sebastianhaas) [inline src/ConsoleClientCredentials.NETCore/README.md]: *typically

File: src/ConsoleClientCredentials.NETCore/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+﻿# Tapkey API Console Sample with Client Credentials (C#, .NET Core 2.2) 
+
+## Client credentials flow
+This console application demonstrates how to consume the Tapkey API from an application using the client credentials flow.
```
Comment (sebastianhaas) [inline src/ConsoleClientCredentials.NETCore/README.md]: *Tapkey Management API

File: src/ConsoleClientCredentials.NETCore/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+## Client credentials flow
+This console application demonstrates how to consume the Tapkey API from an application using the client credentials flow. 
+
+The Client Credentials flow is tipically used when the client (the application) is acting on its own behalf. This means that there's no actual user/person interaction in the process. The client (application) is the owner of the resources.
+
+## What does this application do?
+
+This application will print a list of all locking products that are managed by this OAuth client (as a co-admin in the main owner account)
+
```
Comment (sebastianhaas) [inline src/ConsoleClientCredentials.NETCore/README.md]: double line break

File: README.md [hunk 4]
```diff
@@ -2,12 +2,13 @@
...
-| [TapkeyApiSampleConsole](TapkeyApiSampleConsole%20CSharp%20.NET%20Core%202.0) | C# .NET Core 2.0 | A command-line application that prints a list of a user's locking devices. |
-| [TapkeyApiSampleConsole](TapkeyApiSampleConsole%20Visual%20Basic%20.NET%20Framework%204.6) | Visual Basic .NET Framework 4.6 | A command-line application that prints a list of a user's locking devices. |
+| Name                   | Technology | Grant type | Description |
+|:-----------------------|:-----------|:-----------|:------------|
+| [ConsoleClientCredentials.NETCore](src/ConsoleClientCredentials.NETCore) | C# .NET Core 2.2 | Client Credentials | A command-line application that prints a list of a user's locking devices. |
+| [ConsoleClientCredentials.VB.NetFramework](src/ConsoleClientCredentials.VB.NetFramework) | VB .NET Framework 4.7.2 | Client Credentials | A command-line application that prints a list of a user's locking devices. |
+| [MVCAppAuthorizationCodePKCE.NETCore](src/MVCAppAuthorizationCodePKCE.NETCore) | C# ASP.NET Core MVC 2.2 | Authorization code with PKCE | An ASP.NET MVC web site that prints a list of a user's locking devices. |
```
Comment (sebastianhaas) [inline README.md]: Authorization **C**ode, or Client **c**redentials

**Comment by minichma** [comment]
Looks good code-wise. Suggest to do a little cleanup in the commit history. The merge points probably shouldn't be part of the PR's history.

**Pull request closed without merging**